tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub get_test_name {
+    return my $name = (caller(2))[3];
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
```

Comment:
```suggestion
    my $test_name = $test{'test'} // die('Missing key "test"');
    my $description = $test{'description'} // die('Missing key "description");
    my $result = $test{'result'} // die ('Missing key result');
```
But we don't `die` on extra keys anymore, but I think it is more readable than a `foreach (keys %test)` loop in this case.
If you need that extra key check, I think it is enough to check here for key num.
```
die("Unexpected key") if(keys %test > 3);
```